test/crit/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/checkpoint-restore/go-criu/v5/crit"
+)
+
+func main() {
+	imgs := getImgs()
+	for _, img := range imgs {
+		log.Println("=== ", img)
+		testImg := fmt.Sprintf("%s.test.img", filepath.Base(img))
+		c := crit.New(img, testImg, "", false, false)
+		// Decode the binary image file
+		decodedImg, err := c.Decode()
+		if err != nil {
+			log.Fatal("[DECODE]: ", err)
+		}
+		// Encode it into test binary image file
+		err = c.Encode(decodedImg)
+		if err != nil {
+			log.Fatal("[ENCODE]: ", err)
+		}
+		// Open and compare original and test files
+		imgBytes, err := os.ReadFile(img)
+		if err != nil {
+			log.Fatal(err)
+		}
+		testImgBytes, err := os.ReadFile(testImg)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if !bytes.Equal(imgBytes, testImgBytes) {
+			log.Fatal("[RECODE]: files do not match")
+		}
+		log.Println("=== PASS")
+	}
+}
+
+func getImgs() []string {
+	/* Certain image files generated by CRIU do not
+	use the protobuf format and contain raw binary
+	data. Some image files are also generated using
+	external tools (ifaddr, route, tmpfs). As these
+	images cannot be processed by CRIT, they are
+	excluded from the tests. */
+	var skipImgs = []string{
+		"pages-",
+		"pages-shmem-",
+		"iptables-",
+		"ip6tables-",
+		"nftables-",
+		"route-",
+		"route6-",
+		"ifaddr-",
+		"tmpfs-",
+		"tmpfs-dev-",
+		"autofs-",
+		"netns-ct-",
+		"netns-exp-",
+		"rule-",
+	}
+	var imgs []string
+	dir, err := os.ReadDir(".")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+nextFile:
+	for _, file := range dir {
+		name := file.Name()
+		if filepath.Ext(name) == ".img" {
```

Comment:
we might want to exclude `.json.img`, `.test.img` as well?

Reply:
I had ordered the tests in the makefile so that the integration test ran first, before the E2E test generated any of these files. Is there a possibility we'll run them in the wrong order?

Reply:
Thank you for pointing this out. The idea is for the e2e and integration tests to be able to run independently.
For example, we can run the following:
```
sudo make -C test/crit/ e2e-test
sudo make -C test/crit/ integration-test
sudo make -C test/crit/ integration-test
```
In this case, we have to make sure that integration-test would not process the output files that have been generated from the previous run (`*.json.img`, `*.test.img`, as well as `tmp.*.img` directories).